tests/drop_counters/test_configurable_drop_counters.py

```diff
@@ -0,0 +1,214 @@
+"""
+Tests the functionality of the configurable drop counter feature in SONiC.
+
+Todo:
+    - Add test cases for ACL_ANY, SIP/DIP_LINK_LOCAL, and UNRESOLVED_NEXT_HOP
+    - Add test cases for dynamic add/remove of drop reasons
+    - Add test cases with multiple drop counters
+    - Verify standard drop counters as well as configurable drop counters
+"""
+
+import logging
+import random
+import time
+import pytest
+from natsort import natsorted
+
+import ptf.testutils as testutils
+from netaddr import IPNetwork
+
+import configurable_drop_counters as cdc
+from mock_server import MockServer
+from common.helpers.assertions import pytest_assert
+
+PACKET_COUNT = 1000
+
+@pytest.mark.parametrize("drop_reason", ["L3_EGRESS_LINK_DOWN"])
+def test_neighbor_link_down(testbed_params, setup_counters, duthost, mock_server, test_runner,
+                            drop_reason):
+    """
+    Verifies that counters that check for a neighbor link being down work properly.
+
+    Note:
+        This test works by mocking a server within a VLAN, thus the T0 topology is required.
+
+    Args:
+        drop_reason (str): The drop reason being tested.
+    """
+    counter_type = setup_counters([drop_reason])
+
+    rx_port = random.choice([intf
+                             for intf in testbed_params["ports"]
+                             if intf != mock_server.get_neighbor_iface()])
+    pkt = _get_simple_ip_packet(duthost, rx_port, "2.2.2.2", mock_server.get_addr())
+
+    try:
+        mock_server.start()
+        mock_server.shutdown_link()
+        test_runner(counter_type, pkt, rx_port)
+    finally:
+        mock_server.startup_link()
+        mock_server.shutdown()
+        duthost.command("sonic-clear fdb all")
+        duthost.command("sonic-clear arp")
+
+@pytest.fixture(scope="module")
+def testbed_params(duthost, testbed):
+    """
+    Gathers parameters about the testbed for the test cases to use.
+
+    Returns: A Dictionary with the following information:
+        "ports": A List of the active ports in the system
+        "phy_intfs": A List of ALL physical interfaces in the system
+        "vlan_intfs": A Dictionary containing all the VLAN interfaces in the system, including
+            their address, subnet, prefix length, and VLAN members.
+    """
+
+    if testbed["topo"]["name"] != "t0":
```

Comment:
You probably should check 'type' instead of 'name'. If I am not mistaking, name would be t0-64, t0-116, etc. type would be 't0'?